docs/3.0/custom-widgets-ios.md

```diff
@@ -0,0 +1,112 @@
+
+`+ (NSMutableSet *)remoteObjectProperties` contains all of the widget’s properties that can be accessed by JavaScript. You have to call `super` of this method first. Afterwards add the names of properties defined by your widget to this set, to expose those to JavaScript.
+
+`- (UIView *)view` returns the actual instance of the iOS native widget. You should have property in your class that will be returned by this method. This method does not require a `super` call.
+
+### Calls from the JavaScript
+
+The JavaScript API can invoke methods on the native widget. To do this, you have to register a selector for the specific JavaScript call. Here is an example:
+
+```objc
+- (instancetype)initWithObjectId:(NSString *)aId andClient:(TabrisClient *)tabrisClient {
+  self = [super initWithObjectId:aId andClient:tabrisClient];
+  if (self) {
+    //Prior initialization
+    [self registerSelector:@selector(beep:) forCall:@"beep"];
+    //Following initialization
+  }
+  return self;
+}
+
+- (void)beep:(NSDictionary *)parameters {
+}
+```
+
+If the JavaScript invokes a `beep` method, the call will be forwarded to the native `- (void)beep:(NSDictionary *)parameters` method. Please note that methods you are registering have to accept `NSDictionary` as a parameter. This dictionary contains all of the parameters passed by JavaScript to the native side.
+
+### Events
+
+Events are notifications sent from the native side to JavaScript. In order to add an event to your widget, you only have to declare a public boolean property conforming to this naming pattern: `eventNameListener` (e.g. `longPressListener`). Please note you should not change the value of this property inside of your native code.
+
+```objc
+@interface MyWidget : BasicWidget
+@property (assign) BOOL myEventListener;
+@end
+
+@implementation MyWidget
+
+- (void)someMethod {
+  //Method implementation
+  if (self.myEventListener) {
+    Message<Notification> *message = [[self notifications] forObject:self];
+    [message fireEvent:@"myEvent" withAttributes:@{@"key":@"value", @"otherKey":@"otherValue"}];
+  }
+}
+```
+
+### Registering your widget
+
+You have to execute following instructions after you add a plugin (e.g. `cordova plugin add`) to your project. Locate your Xcode project file (typically located in `platforms/ios/`) and open it.
+
+The final step of this process is adding your new widget to widget registry. To do this edit the `AppDelegate.m` in your project to:
+
+* Import your plugins header file (e.g. `#import "MyWidget.h"`)
+* Find the `- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions` method. Inside of this method you have add following line `[self.client addRemoteObject:[MyWidget class]];` along the other calls of `addRemoteObject:` method.
+
```

Comment:
Please suggest a new text and what lines it replaces.

Reply:
Afaik adding this to your plugin.xml should be enough:

```xml
 <platform name="ios">
    <config-file target="*TabrisPlugins.plist" parent="classes">
      <array>
        <string>MyWidget</string>
      </array>
    </config-file>
  </platform>
```

This would replace the whole _Registering your widget_ section.

@karolszafranski @patrykmol Please correct me if I'm wrong.